search-handlers/src/main/java/no/unit/nva/indexing/handlers/IndexResourceHandler.java

```diff
@@ -39,18 +41,18 @@ public static IndexingClient defaultIndexingClient() {
     }
 
     @Override
-    protected Void processInputPayload(IndexEvent input, AwsEventBridgeEvent<AwsEventBridgeDetail<IndexEvent>> event,
+    protected Void processInputPayload(EventReference input,
+                                       AwsEventBridgeEvent<AwsEventBridgeDetail<EventReference>> event,
                                        Context context) {
 
         UnixPath resourceRelativePath = new UriWrapper(input.getUri()).toS3bucketPath();
-        String resource = s3Driver.getFile(resourceRelativePath);
-        IndexDocument indexDocument = IndexDocument.fromJsonString(resource);
-        try {
-            indexingClient.addDocumentToIndex(indexDocument);
-        } catch (SearchException e) {
```

Comment:
removed the part where an IOException is put in SearchException which is put in a RuntimeException.